File: Algorithms/Sortings/Go/selectionsort.go
func SelectionSort(arr T[]) T[] {
	for i := 0; i < len(arr) - 1; i++ {
		min := 0
		for j := i + 1; j < len(arr); j++ {
			if(arr[j] < arr[min]) {
				min = j
			}
		}
		if(min != i) {
			arr[i], arr[min] = arr[min], arr[i]
		}
	}
	return arr
}

File: Algorithms/Sortings/JavaScript/selection-sort.js
const arr = [4, -3, 3, 18, 2, 7, 55, 0];

for (let i = 0; i < arr.length - 1; i++) {
  let min = i;
  for (let j = i + 1; j < arr.length; j++) {
    if (arr[min] > arr[j]) {
      min = j;
    }
  }
  if (min != i) {
    [arr[i], arr[min]] = [arr[min], arr[i]];
  }
}

console.log(arr);
